botx/models/attachments.py

```diff
@@ -0,0 +1,262 @@
+
+    #: file content in RFC 2397 format.
+    content: str  # noqa: WPS110
+
+    #: voice duration
+    duration: int
+
+
+class Location(BaseModel):
+    """Location model from botx."""
+
+    #: name of location
+    location_name: str
+
+    #: address of location
+    location_address: str
+
+    #: latitude of location
+    location_lat: float
+
+    #: longitude of location
+    location_lng: float
+
+
+class Contact(BaseModel):
+    """Contact model from botx."""
+
+    #: name of contact
+    contact_name: str
+
+
+class Link(BaseModel):
+    """Class that marked as Link from botx."""
+
+    #: url of link
+    url: str
+
+    #: title of url
+    url_title: Optional[str] = None
+
+    #: link on preview of this link
+    url_preview: Optional[str] = None
+
+    #: text on preview
+    url_text: Optional[str] = None
+
+    def is_mail(self) -> bool:
+        """Confirm that is email link."""
+        return self.url.startswith(MAIL_PROTO_STR)
+
+    def is_telephone(self) -> bool:
+        """Confirm that is telephone link."""
+        return self.url.startswith(TELEPHONE_PROTO_STR)
+
+    def is_link(self) -> bool:
+        """Confirm that is link on resource."""
+        return not (self.is_mail() or self.is_telephone())
+
+    @property
+    def mailto(self) -> str:
+        """Property that retuning email address without protocol."""
+        if not self.is_mail():
+            raise MissingAttributeError(attribute_name="mailto")
+        # fmt: off
+        return self.url[len(MAIL_PROTO_STR):]
+        # fmt: on
+
+    @property
+    def tel(self) -> str:
+        """Property that retuning telephone number without protocol."""
+        if not self.is_telephone():
+            raise MissingAttributeError(attribute_name="telephone number")
+        # fmt: off
+        return self.url[len(TELEPHONE_PROTO_STR):]
+        # fmt: on
+
+
+AttachmentMapping = {
+    AttachmentsTypes.image: Image,
+    AttachmentsTypes.video: Video,
+    AttachmentsTypes.document: Document,
+    AttachmentsTypes.voice: Voice,
+    AttachmentsTypes.contact: Contact,
+    AttachmentsTypes.location: Location,
+    AttachmentsTypes.link: Link,
+}
+
+Attachments = Union[
+    Video, Voice, Contact, Location, Link, Image, Document,
+]
+
+
+class Attachment(BaseModel):
+    """Attachment with main attach and type of this attachment."""
+
+    #: type of attachment
+    type: str  # noqa: WPS125
+
+    #: content of attachment
+    data: Attachments  # noqa: WPS110
+
+
+class AttachList(BaseModel):  # noqa: WPS214
+    """Additional wrapped class for use property."""
+
+    __root__: List[Attachment]
+
+    @property
+    def image(self) -> Image:
+        """Parse attachments.
+
+        Return:
+            Image.
+        """
+        for attach in self.__root__:
+            if attach.type == AttachmentsTypes.image:
+                return attach.data  # type: ignore
+        raise MissingAttributeError(attribute_name=AttachmentsTypes.image)
```

Comment:
move this into private method instead of duplicating code and leave only definitions with casts. something like this:
```python3
@property
def image(self) -> Image:
    attach = self._get_attachment_by_type(AttachmentsTypes.image)
    return cast(Image, attach)
```